server/mediaserver.go

```diff
@@ -291,8 +291,14 @@ func createRTMPStreamIDHandler(s *LivepeerServer) func(url *url.URL) (strmID str
 			return nil
 		}
 		if _, exists := s.rtmpConnections[mid]; exists {
-			glog.Errorf("Manifest manifestID=%v already exists for streamID url=%s", mid, url.String())
-			return nil
+			if extmid != mid {
+				// only error out if we're seeing segments from different extmids but same mid trying to use the same connection
+				// FIXME s.internalManifests might not be updated if segments are too close together
+				if _, extmidSeenBefore := s.internalManifests[extmid]; !extmidSeenBefore {
+					glog.Errorf("RTMP connection for manifest manifestID=%v already exists (streamID url=%s)", mid, url.String())
+					return nil
```

Comment:
I think instead of returning error here we should add `ExtrenalManifestID` field to the `core.StreamParameters` struct, and based on that info push handler will be able to remove old session with same playbackID and replace it with the new one